pkg/kubelet/images/image_manager.go

```diff
@@ -124,23 +136,23 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, objRef *v1.ObjectR
 		RuntimeHandler: podRuntimeHandler,
 	}
 
-	imageRef, err = m.imageService.GetImageRef(ctx, spec)
+	imageRef, err = m.imagePullPrecheck(ctx, pullPolicy, spec)
 	if err != nil {
```

Comment:
I think the whole errors recording logic can be moved in the `imagePullPrecheck` as well. You will be able to test it using fakeRecorder than

Reply:
Tried to test this by adding some new fakeRecorder methods for comparing an expected list of events to an observed list of events